pkg/controllers/subnetport/addressbinding_webhook.go

```diff
@@ -56,8 +57,24 @@ func (v *AddressBindingValidator) Handle(ctx context.Context, req admission.Requ
 			log.Error(err, "error while decoding AddressBinding", "AddressBinding", req.Namespace+"/"+req.Name)
 			return admission.Errored(http.StatusBadRequest, err)
 		}
-		if !reflect.DeepEqual(ab.Spec, oldAddressBinding.Spec) {
-			return admission.Denied("update AddressBinding is not allowed")
+		if ab.Spec.VMName != oldAddressBinding.Spec.VMName || ab.Spec.InterfaceName != oldAddressBinding.Spec.InterfaceName {
+			return admission.Denied("update AddressBinding vmName/interfaceName is not allowed")
+		}
+	}
+	if ab.Spec.IPAddressAllocationName != "" {
+		ipAllocation := &v1alpha1.IPAddressAllocation{}
+		if err := v.Client.Get(context.TODO(), types.NamespacedName{
+			Namespace: ab.Namespace,
+			Name:      ab.Spec.IPAddressAllocationName,
+		}, ipAllocation); err != nil {
+			log.Error(err, "failed to get IPAddressAllocation", "IPAddressAllocation", ab.Namespace+"/"+ab.Spec.IPAddressAllocationName)
+			return admission.Denied("ipAddressAllocationName is not valid")
```

Comment:
Let us report `"IPAddressAllocation %s does not exist", ipAddressAllocationName`

Reply:
Fixed.